staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -271,89 +333,105 @@ func handleSliceDiff(key string, originalValue, modifiedValue []interface{}, pat
 			patch[parallelDeletionListKey] = deletionList
 		}
 	default:
-		replacePatchFieldIfNotEqual(key, originalValue, modifiedValue, patch, diffOptions)
+		changed = replacePatchFieldIfNotEqualOrReplaceKeys(key, originalValue, modifiedValue, patch, diffOptions)
 	}
-	return nil
+	return changed, nil
 }
 
-// replacePatchFieldIfNotEqual updates the patch if original and modified are not deep equal
+// replacePatchFieldIfNotEqualOrReplaceKeys updates the patch if original and modified are not deep equal
```

Comment:
Maybe not rename this because it's too long and doesn't cover IgnoreChangesAndAdditions anyways